.github/workflows/test.yml

```diff
@@ -63,10 +69,37 @@ jobs:
         uses: Swatinem/rust-cache@v2
       - name: Setup coverage env
         uses: taiki-e/install-action@cargo-llvm-cov
+      # Python setup
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Load cached Poetry installation
+        id: cached-poetry
+        uses: actions/cache@v3
+        with:
+          path: ~/.local
+          key: poetry-${{ runner.os }}
+      - name: Install Poetry
+        if: steps.cached-poetry.outputs.cache-hit != 'true'
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+      - name: pull kakarot submodule and setup
+        run:
+          make setup
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+      - run: cd kakarot && poetry config installer.modern-installation false
       - name: Generate code coverage
         run:
-          cargo llvm-cov --all-features --workspace --lcov --output-path
-          lcov.info
+          ./scripts/make_with_env.sh test-coverage
```

Comment:
could this be refactored with:

`env` yaml key in your job in order to source .env variables without using a .sh script?

Reply:
example from kakarot:

```
jobs:
  deploy:
    runs-on: ubuntu-latest
    env:
      ACCOUNT_ADDRESS: ${{ secrets.ACCOUNT_ADDRESS }}
      PRIVATE_KEY: ${{ secrets.PRIVATE_KEY }}
      DEVNET_ACCOUNT_ADDRESS: ${{ secrets.DEVNET_ACCOUNT_ADDRESS }}
      DEVNET_PRIVATE_KEY: ${{ secrets.DEVNET_PRIVATE_KEY }}
      RPC_KEY: ${{ secrets.RPC_KEY }}
      INFURA_KEY: ${{ secrets.INFURA_KEY }}
    outputs:
      devnet-version: ${{ steps.devnet-version.outputs.DEVNET_VERSION }}
    steps:
      - uses: actions/checkout@v3
// etc.
```

Reply:
PS: does this need "make" before test-coverage, or all make commands can just be accessed like this?

Reply:
all commands.

the only env variable we need is for testing, so i dont think it makes sense to establish it as a secret. i can remove the env variable and just hardcode that the tests will expect deploy artifacts in the ./katana/build directory. i think that may be the right way forward.

Reply:
alright works for me